File: 프로그래머스/1/87389. 나머지가 1이 되는 수 찾기/나머지가 1이 되는 수 찾기.java
class Solution {
    public int solution(int n) {
        int answer= 0;
        
        for(int i=n-1; i>0; i--){       
            answer = n % i == 1 ? i : answer;
        }
        return answer;
    }
}

// import java.util.stream.IntStream;

// class Solution {
//     public int solution(int n) {
//         return IntStream.range(2, n).filter(i -> n % i == 1).findFirst().orElse(0);
//     }
// } 스트림 공부용

File: 프로그래머스/1/87389. 나머지가 1이 되는 수 찾기/나머지가 1이 되는 수 찾기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int n) {
    int answer = 0;
    
    for(int i=n-1; i>0; i--){ 
        answer = n % i == 1 ? i : answer;
    }   
    
    return answer;
}